src/Illuminate/Routing/PendingResourceRegistration.php

```diff
@@ -144,32 +144,41 @@ public function parameter($previous, $new)
     /**
      * Add middleware to the resource routes.
      *
-     * @param  mixed  $middleware
+     * @param mixed $middleware
      * @return \Illuminate\Routing\PendingResourceRegistration
      */
     public function middleware($middleware)
     {
         $middleware = Arr::wrap($middleware);
+        $method_middleware = [];
+
+        foreach ($middleware as $key => $value) {
+            if (is_string($key)) {
+                $method_middleware[$key] = Arr::wrap($value);
+                unset($middleware[$key]);
+            }
+        }
```

Comment:
The change.

Kinda gross. Should also becasting the wrapped array values to strings. Again, general API is the main concern righ tnow